packages/siwe/src/mapToSIWX.ts

```diff
@@ -183,15 +183,22 @@ export function mapToSIWX(siwe: AppKitSIWEClient): SIWXConfig {
         }
 
         const siweSession = await getSession()
-        const siweCaipNetworkId = `eip155:${siweSession?.chainId}`
+        const siweCaipNetworkId = `${ConstantsUtil.CHAIN.EVM}:${siweSession?.chainId}`
+
         const lowercaseSessionAddress = siweSession?.address?.toLowerCase()
-        const lowercaseCaipAddress = address?.toLowerCase()
+        const caipSessionAddress = `${siweCaipNetworkId}:${siweSession?.address}`
+
+        const lowercaseAddress = address?.toLowerCase()
+        const caipAddress = `${chainId}:${address}`
+
+        const signOutOnNetworkChange = siwe.options.signOutOnNetworkChange
+
+        const shouldSignOut = signOutOnNetworkChange
+          ? caipSessionAddress !== caipAddress
+          : lowercaseSessionAddress !== lowercaseAddress
```

Comment:
We can use this function here ? https://github.com/reown-com/appkit/blob/661548e3b7e2b8650935a3af22303ae3cf685111/packages/appkit-utils/src/HelpersUtil.ts#L19